src/main/java/com/internetitem/logback/elasticsearch/AbstractElasticsearchPublisher.java

```diff
@@ -166,6 +175,9 @@ private void serializeEvents(JsonGenerator gen, List<T> eventsCopy, List<Abstrac
 			gen.writeRaw('\n');
 			serializeEvent(gen, event, propertyList);
 			gen.writeRaw('\n');
+			gen.flush();
+			outputAggregator.write(temporaryWriter.getBuffer().toString());
+			temporaryWriter.getBuffer().setLength(0);
```

Comment:
When an event is fully generated by the JSONGenerator, we take the resulting string as One String, and write it to outputAggregator. If it fits it fits, if not, no fragments will be in the request.

Since the generator was hooked up to outputAggregator directly before, the payload could theoretically end with any JSON fragment. So it wasn't just the newline missing as we suspected, but if the buffer is full, all kinds of crap.